File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Blindness.java

package com.watabou.pixeldungeon.actors.buffs;

import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.ui.BuffIndicator;

import org.jetbrains.annotations.NotNull;

public class Blindness extends FlavourBuff {


	@Override
	public boolean attachTo(@NotNull Char target) {
		boolean ret = super.attachTo(target);
		if (ret) {
			target.observe();
		}
		return ret;
	}

	@Override
	public void detach() {
		super.detach();
		target.observe();
	}
	
	@Override
	public int icon() {
		return BuffIndicator.BLINDNESS;
	}
}

File: scripts/buffs/Blindness.lua
---
--- Blindness: shrinks view distance while active (was actors/buffs/Blindness.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = 16, -- BuffIndicator.BLINDNESS
            name          = "BlindnessBuff_Name",
            info          = "BlindnessBuff_Info",
        }
    end,

    attached = function(self, buff)
        buff.target:observe()
    end,

    detach = function(self, buff)
        buff.target:observe()
    end
}
